components/job-orchestration/job_orchestration/executor/compress/fs_compression_task.py

```diff
@@ -212,48 +238,61 @@ def run_clp(
 
     # Compute the total amount of data compressed
     last_archive_stats = None
+    last_line_decoded = False
     total_uncompressed_size = 0
     total_compressed_size = 0
-    while True:
+
+    # Handle job metadata update and s3 write if enabled
+    s3_error = None
+    while not last_line_decoded:
         line = proc.stdout.readline()
-        if not line:
-            break
-        stats = json.loads(line.decode("ascii"))
-        if last_archive_stats is not None and stats["id"] != last_archive_stats["id"]:
-            # We've started a new archive so add the previous archive's last
-            # reported size to the total
-            total_uncompressed_size += last_archive_stats["uncompressed_size"]
-            total_compressed_size += last_archive_stats["size"]
-            with closing(sql_adapter.create_connection(True)) as db_conn, closing(
-                db_conn.cursor(dictionary=True)
-            ) as db_cursor:
-                update_job_metadata_and_tags(
-                    db_cursor,
-                    job_id,
-                    clp_metadata_db_connection_config["table_prefix"],
-                    tag_ids,
-                    last_archive_stats,
-                )
-                db_conn.commit()
+        stats: Optional[Dict[str, Any]] = None
+        if line:
+            stats = json.loads(line.decode("ascii"))
+        else:
+            last_line_decoded = True
+
+        if last_archive_stats is not None and (
+            None is stats or stats["id"] != last_archive_stats["id"]
+        ):
+            if enable_s3_write:
+                archive_id = last_archive_stats["id"]
+                src_archive_file = archive_output_dir / archive_id
+
+                if s3_error is None:
+                    logger.info(f"Uploading archive {archive_id} to S3...")
+                    result = s3_put(s3_config, src_archive_file, archive_id)
+
+                    if result.is_err():
+                        logger.error(f"Failed to upload archive {archive_id}: {result.err_value}")
+                        s3_error = result.err_value
+                        # Note: it's possible proc finishes before we call terminate() on it. In
+                        # Which case the process will still return success.
+                        proc.terminate()
+                    else:
+                        logger.info(f"Finished uploading archive {archive_id} to S3.")
+
+                src_archive_file.unlink()
+
```

Comment:
The file is deleted even if the upload fails -> this is expected.
The process termination might not be immediate -> I think it's ok. the process can keep running since the loop will anyway clean up the generated archives.